### PR TITLE
Hadoop streaming command options

### PR DESCRIPTION
**Pull request by heuvel**
Added support for getting the archives and cmdenvs parameters from the luigi configuration file and setting the environment variables for hadoop streaming jobs.

### CODE REVIEW

File: luigi/contrib/hadoop.py [hunk 1]
```diff
@@ -582,7 +587,15 @@ class DefaultHadoopJobRunner(HadoopJobRunner):
     def __init__(self):
         config = configuration.get_config()
         streaming_jar = config.get('hadoop', 'streaming-jar')
-        super(DefaultHadoopJobRunner, self).__init__(streaming_jar=streaming_jar)
+        cmdenvs = config.get('hadoop', 'cmdenvs', default=None)
+        archives = config.get('hadoop', 'archives', default=None)
```
Comment (mfcabrera) [inline luigi/contrib/hadoop.py]: My personal opinion: Archives options was already added as part of  https://github.com/spotify/luigi/pull/1649/files , I don't think we should force people to use the configuration file. Users can simple extended this.  `HadoopJobRunner` or `DefaultHadoopJobRunner` to get this done. In particular users already doing this programmatically will have the code broken  by this (my case).

Comment (heuvel) [inline luigi/contrib/hadoop.py]: I agree, so I have removed that part in the new commit.

**Comment by erikbern** [comment]
lgtm. maybe it should just be an argument for extra arguments

**Comment by Tarrasch** [comment]
Can this be tested?

**Comment by heuvel** [comment]
The use case for this is that we have Python 2.7 installed on each cluster node, but we are not allowed to install additional packages. So I create an archive of the virtual environment with the required packages (as described here [http://mfcabrera.com/python/2016/04/25/python-streaming-blog-org.html](url) by @mfcabrera) and I create a custom hadoopjobrunner to load the archive and environment variables from the configuration file. The cmdenv option of hadoop streaming is used to add the packages in the virtual environment archive to the Python path.

The HadoopJobRunner:

    class AdaHadoopJobRunner(luigi.contrib.hadoop.HadoopJobRunner):
    
        def __init__(self):
            config = configuration.get_config()
            streaming_jar = config.get('hadoop', 'streaming-jar')
            cmdenvs = config.get('hadoop', 'cmdenvs', default=None)
            archives = config.get('hadoop', 'archives', default=None)
            
            if cmdenvs:
                cmdenvs = cmdenvs.split(';')
            
            if archives:
                archives = archives.split(';')
                
            super(AdaHadoopJobRunner, self).__init__(streaming_jar=streaming_jar, cmdenvs=cmdenvs, archives=archives)


The luigi.cfg:

	...
	
	[hadoop]
	...
	cmdenvs=PYTHONPATH=venv_archive/lib/python2.7/site-packages/
	archives=venv_archive_with_packages.zip#venv_archive;

**Comment by mfcabrera** [comment]
@heuvel  I create a PR with a more general version of this: #1929    Maybe you could try to adapt your code? . I think is cleaner than adding  rarely used parameters.  You will only need to add the  `extra_arguments` method to a normal JobTask.

**Comment by heuvel** [comment]
@mfcabrera The more general version as proposed in #1929 will definitely work for me, so I will close this PR. Thank you!

**Comment by mfcabrera** [comment]
@heuvel Cool!  can you give it a try (just checkout that branch). We actually need someone else using Streaming jobs to confirm it works and then merge into master. :)

**Pull request closed without merging**